client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -318,9 +319,13 @@ export default React.createClass( {
 	},
 
 	recordSubmit() {
+		const errors = formState.getErrorMessages( this.state.form );
 		analytics.tracks.recordEvent( 'calypso_contact_information_form_submit', {
-			errors: formState.getErrorMessages( this.state.form )
+			errors,
+			errors_count: errors && errors.length || 0,
+			submission_count: this.state.submissionCount + 1
 		} );
+		this.setState( { submissionCount: this.state.submissionCount + 1 } );
```

Comment:
`setState` is not guaranteed to execute immediately which results in us sending the event before updating the numbers :)